### PR TITLE
Update I2C API

### PR DESCRIPTION
**Pull request by scottgonzalez**
The only component I was able to test with a real device was `wii`, but the API change is very straight-forward.

### CODE REVIEW

**Comment by Resseguie** [comment]
From @frxnz: "Just tried to test IR with a GP2Y0D805Z0F. Technically it behaves the same, but it looks like there's a bug in master" (See #766)

**Comment by rwaldron** [comment]
I'll try a few devices, but I'm pretty sure this is totally fine.

File: lib/ir.js [hunk 1]
```diff
@@ -146,20 +146,20 @@ function IR(opts) {
...
   setInterval(function() {
     // Set pointer to X most signficant byte/register
-    this.io.sendI2CWriteRequest(address, preread);
+    this.io.i2cWrite(address, preread);
 
     // Read from register
-    this.io.sendI2CReadRequest(address, bytes, data.bind(this));
+    this.io.i2cReadOnce(address, bytes, data.bind(this));
 
   }.bind(this), delay);
```
Comment (rwaldron) [inline lib/ir.js]: Confirmed, this is ok. 

Also, this file has reached end of life.

File: lib/led/ledcontrol.js [hunk 2]
```diff
@@ -540,7 +540,7 @@ Controllers = {
       if (arguments.length !== 3) {
         throw new Error("`send` expects three arguments: device, opcode, data");
       }
-      this.io.sendI2CWriteRequest(this.addresses[addr], [opcode | data]);
+      this.io.i2cWrite(this.addresses[addr], [opcode | data]);
       return this;
     }
   },
```
Comment (rwaldron) [inline lib/led/ledcontrol.js]: Confirmed, this is ok.

File: lib/sonar.js [hunk 3]
```diff
@@ -129,7 +129,7 @@ Devices = {
 
       function prime() {
         // 0x52 result in us (microseconds)
-        this.io.sendI2CWriteRequest(address, [0x00, 0x52]);
+        this.io.i2cWrite(address, [0x00, 0x52]);
 
         setTimeout(read.bind(this), delay);
       }
```
Comment (rwaldron) [inline lib/sonar.js]: Confirmed, this is ok.

Also, this file has reached end of life.

**Comment by scottgonzalez** [comment]
Sounds like this is good to land then?

**Comment by Resseguie** [comment]
LGTM

**Comment by rwaldron** [comment]
I haven't reality checked the Wii stuff, but you have, so that covers everything!

**Pull request merged**